modules/core/src/lib/layer.js

```diff
@@ -418,6 +418,10 @@ export default class Layer extends Component {
     }
   }
 
+  _onViewportChange() {
```

Comment:
This private virtual function is very subtle (in that it is not clear from a reading of either file that these are related). I guess the name `_on...` is a hint but I didn't get it until I reread the code.

I wish it could be handled through an existing flags / update call mechanism but if not this is fine.

Maybe add a brief comment in each place explaining the relation.